lib/Star/Kata/KataDomain/KataService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\KataDomain;
+
+use Star\Kata\Model\KataCollection;
+use Star\Kata\Model\KataRepository;
+
+/**
+ * Class KataService
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\KataDomain
+ */
+final class KataService
+{
+    const CLASS_NAME = __CLASS__;
+
+    /**
+     * @var KataRepository
+     */
+    private $collection;
+
+    /**
+     * @param KataRepository $repository
+     */
+    public function __construct(KataRepository $repository)
+    {
+        $this->collection = $repository;
+    }
+
+    /**
+     * @param $kataName
+     *
+     * @return \Star\Kata\Model\StartedKata
+     * @throws \RuntimeException
+     */
+    public function startKata($kataName)
+    {
+        if (empty($kataName)) {
+            throw new \RuntimeException('Kata must be supplied.');
+        }
+
+        $kata = $this->collection->getKata($kataName);
+        if (null === $kata) {
+            throw new \RuntimeException("The '{$kataName}' kata was not found.");
+        }
+
+        return $kata->start();
```

Comment:
what does start a `Kata` actually do? I would expect it to just change the `Kata`'s state and `return` `void`.